test/interpreter/function-npv.spec.ts

```diff
@@ -0,0 +1,42 @@
+import {ErrorType, HyperFormula} from '../../src'
+import {ErrorMessage} from '../../src/error-message'
+import {adr, detailedError} from '../testUtils'
+
+describe('Function NPV', () => {
+  it('should return #NA! error with the wrong number of arguments', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=NPV(1)'],
+    ])
+
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA, ErrorMessage.WrongArgNumber))
+  })
+
+  it('should return correct value', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=NPV(1, NA())'],
+      ['=NPV(1, B2:C2)', 1, 'abcd'],
+      ['=NPV(1, B3:C3)', true, 1],
+      ['=NPV(1, 1, "abcd")'],
+      ['=NPV(1, TRUE(), 1)'],
+      ['=NPV(-1, 0)'],
+      ['=NPV(-1,1)'],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqual(detailedError(ErrorType.NA))
+    expect(engine.getCellValue(adr('A2'))).toEqual(0.5)
+    expect(engine.getCellValue(adr('A3'))).toEqual(0.5)
+    expect(engine.getCellValue(adr('A4'))).toEqual(detailedError(ErrorType.VALUE, ErrorMessage.NumberCoercion))
+    expect(engine.getCellValue(adr('A5'))).toEqual(0.75)
+    expect(engine.getCellValue(adr('A6'))).toEqual(0)
+    expect(engine.getCellValue(adr('A7'))).toEqual(detailedError(ErrorType.NUM, ErrorMessage.NaN))
+  })
+
```

Comment:
- Can we add a test with a range 3x3? There is this order condition we should check "Because the result is affected by the order of values, evaluators shall evaluate arguments in the order given and range reference and array arguments row-wise starting from top left." [1]

I think other conditions are here but not that clearly described

Source: 
[1] https://docs.oasis-open.org/office/OpenDocument/v1.3/csprd02/part4-formula/OpenDocument-v1.3-csprd02-part4-formula.html#__RefHeading__1018290_715980110
[2] https://support.microsoft.com/en-us/office/npv-function-8672cb67-2576-4d07-b67b-ac28acf2a568

Reply:
Thx 👍